x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.ts

```diff
@@ -260,51 +281,51 @@ The Kibana Connector in use may need to be reconfigured with an updated Amazon B
     model,
   }: InvokeAIActionParams): Promise<InvokeAIActionResponse> {
     const res = await this.runApi({
-      body: JSON.stringify(formatBedrockBody({ messages, model })),
+      body: JSON.stringify(formatBedrockBody({ messages })),
       model,
     });
     return { message: res.completion.trim() };
   }
 }
 
 const formatBedrockBody = ({
-  model = DEFAULT_BEDROCK_MODEL,
   messages,
-  stopSequences = ['\n\nHuman:'],
+  stopSequences,
   temperature = 0.5,
 }: {
-  model?: string;
   messages: Array<{ role: string; content: string }>;
   stopSequences?: string[];
   temperature?: number;
-}) => {
-  const combinedMessages = messages.reduce((acc: string, message) => {
-    const { role, content } = message;
-    const [, , modelName, majorVersion, minorVersion] =
-      (model || '').match(/(\w+)\.(.*)-v(\d+)(?::(\d+))?/) || [];
-    // Claude only has Assistant and Human, so 'user' will be converted to Human
-    let bedrockRole: string;
-
-    if (
-      role === 'system' &&
-      modelName === 'claude' &&
-      Number(majorVersion) >= 2 &&
-      Number(minorVersion) >= 1
-    ) {
-      bedrockRole = '';
-    } else {
-      bedrockRole = role === 'assistant' ? '\n\nAssistant:' : '\n\nHuman:';
-    }
+}) => ({
+  anthropic_version: 'bedrock-2023-05-31',
+  messages: messages.reduce(
+    (acc: Array<{ role: string; content: string }>, m) => [
+      ...acc,
+      ...(m.role === 'system'
+        ? // Bedrock only accepts assistant and user roles.
+          // If a system message is sent, transform it to a user message
+          // followed by an assistant message (as 2 user messages in a row are not allowed)
+          [
+            { ...m, role: 'user' },
+            { content: 'Ok.', role: 'assistant' },
+          ]
```

Comment:
@dgieselaar This is the code I was talking about that would turn your system message into a user message + assistant message